modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/util/utils.js

```diff
@@ -215,6 +190,27 @@ export function removePathTrailingSlash(path) {
 	return path;
 }
 
+/**
+ * Destroys all rendered portlets on the page
+ */
+export function resetAllPortlets() {
+	getPortletBoundaryIds(Liferay.Portlet.list).forEach((value) => {
+		const portlet = document.querySelector('#' + value);
```

Comment:
If you have an ID, [don't make a string-based selector](https://github.com/liferay/liferay-frontend-guidelines/issues/19). Just use `getElementById()`.